src/openapi/batch-util.js

```diff
@@ -25,10 +26,10 @@ const LOG_AREA = 'batch';
  * Parses the response from a batch call.
  * @name saxo.openapi.batchUtil.parse
  * @param {string} responseText
+ * @param {Number} parentRequestId - The parent request id. Used as a base for calculating batch items request ids.

```

Comment:
wouldn't it be simpler to just always ask for a request id form the global request id counter?

Reply:
its for parsing?

Reply:
It's just for parsing and deriving local index from global x-request-id of each sub response, which have strict relation to parent request-id. This ensures same order of batched responses for request items.

I've tried to follow requirement of previous implementation for safety. 

It might be worth simplifying it further, for example if we can get confirmation from openapi that they ensure correct response order to match request order. From what I've saw, it seems to be true. If that's the case, we might drop parentRequestId and x-request-id all together in this parser. Will talk to them about it. 

However, from the existing implementation, it looks like there was a reason for using x-request-id for index of requestData mapping. Which implies that order might change.

Reply:
> wouldn't it be simpler to just always ask for a request id form the global request id counter?

I do exactly that for building requests. Check ```build``` method in this file (batch-util). 
There is no additional complexity for building other then making sure we get x-request-id for parent batched request header before we build children.